util/docker.js

```diff
@@ -0,0 +1,187 @@
+import { confirm, input } from "@inquirer/prompts";
+
+function validatePortFormat(input) {
+  const validFormat = /^\d+:\d+$/;
+  return validFormat.test(input) || "Ports must be in the format hostPort:containerPort.";
+}
+
+async function promptForEnvironmentVariables() {
+  const addEnvironmentVariables = await confirm({
+    message: "Do you want to add environment variables?",
+    default: false,
+  });
+
+  if (addEnvironmentVariables) {
+    const variableInput = await input({
+      message: "Enter environment variables separated by commas (e.g., VAR1=DB_USER, VAR2=DB_PASS):",
+      validate: (input) => {
+        // Regex Expression to Evaluate User Input in Format: VAR1=DB_USER, VAR2=DB_PASS.
+        const validFormat = /^([a-zA-Z_][a-zA-Z0-9_]*=[a-zA-Z_][a-zA-Z0-9_]*)(,\s*[a-zA-Z_][a-zA-Z0-9_]*=[a-zA-Z_][a-zA-Z0-9_]*)*$/;
+        return (
+          validFormat.test(input) ||
+          "Input must be in the format VAR1=DB_USER, VAR2=DB_PASS (comma-separated pairs)."
+        );
+      },
+    });
+
+    const environmentVariables = {};
+    variableInput.split(",").forEach((pair) => {
+      const [key, envVariable] = pair.split("=").map((item) => item.trim());
+      environmentVariables[key] = envVariable;
+    });
+
+    return environmentVariables;
+  }
+
+  return undefined;
+}
+
+async function promptAppService(packageName) {
+  const image = await input({
+    message: "Enter the Docker image for the app service (default: node:20-alpine):",
+    default: "node:20-alpine",
+  });
+
+  const ports = await input({
+    message: "Enter the ports to expose for the app service (default: 8080:8080):",
+    default: "8080:8080",
+    validate: validatePortFormat,
+  });
+
+  const environment = await promptForEnvironmentVariables();
+
+  return {
+    name: packageName,
+    image,
+    containerName: `${packageName}_container`,
+    ports: [ports],
+    environment,
+  };
+}
+
+async function promptDbService(packageName) {
+  const image = await input({
+    message: "Enter the Docker image for the database (default: mysql:latest):",
+    default: "mysql:latest",
+  });
```

Comment:
Can you automatically do this?
Say, we store this in metadata of the templates inside of `bin/configs.js`. We can take it from there.
Even the db-ports. We can configure it from there. Reason being, `mysql` and `pg` that are there right now have default ports right. In future, it is the job of the template creator to mention the port in config.